e2e_tests/suite/test_smoke.py

```diff
@@ -0,0 +1,114 @@
+import time
+
+import pytest
+import requests
+import json
+
+from botocore.waiter import WaiterModel, create_waiter_with_client
+
+from e2e_tests.settings import NGINX_API_VERSION, RECONFIGURATION_DELAY
+
+
+def wait_for_changes_in_api(req_url, desired_capacity) -> None:
+    resp = requests.get(req_url)
+    nginx_upstream = json.loads(resp.text)
+    counter = 0
+    while len(nginx_upstream['peers']) != desired_capacity and counter < 10:
+        time.sleep(RECONFIGURATION_DELAY)
```

Comment:
we use the same delay `RECONFIGURATION_DELAY` here and in the `scale_aws_group` function. Should it be the same ?

`wait_for_changes_in_api` is called after the test already ensure that the autoscaling group has been updated and its instances are in service. This means that NGINX Plus should be reconfigured by that time + the `sync_interval_in_seconds: ?` in the asg sync YAML, right?

Reply:
> Should it be the same?
No, `scale_aws_group` takes ~5 sec whilst actual scaling (starting up the instances) can take up to several minutes. As for delay in `wait_for_changes_in_api` it is equal to RECONFIGURATION_DELAY for stability. I encountered several failures with a smaller delay, maybe it's a place for further investigation but not for the tests.